top5.py

```diff
@@ -0,0 +1,160 @@
+            return None
+        else:
+            print("Find image for ", first_name, " ", last_name)
+            return URL
+    except:
+        print("get_player_image error")
+        return None
+
+
+def update_database(top5_list):
+    # This function is to update the database
+    for player in top5_list:
+        image = get_player_image(player['first_name'], player['last_name'])
+        if image:
+            player_stat = PlayerStats(season=player['season'], first_name=player['first_name'],  last_name=player['last_name'], pts=player['pts'],
+                                      reb=player['reb'], ast=player['ast'], stl=player['stl'], blk=player['blk'], rank=player['rank'], title=player['title'], image=image)
+        else:
+            player_stat = PlayerStats(season=player['season'], first_name=player['first_name'],  last_name=player['last_name'], pts=player['pts'],
+                                      reb=player['reb'], ast=player['ast'], stl=player['stl'], blk=player['blk'], rank=player['rank'], title=player['title'], image=DEFAULT_IMAGE)
+        db.session.add(player_stat)
+        db.session.commit()
+
+
+def sleep_timer(i):
+    # this timer stops for 70 secs after every 44 requests. This is due to the limit of the amount of requests per minute of the API site.
+    while i > 44:
+        time.sleep(70)
+        i = 0
+    return i
+
+
+def top5_list_sort(cat, input_list, first_name, last_name):
+    # sort the top5 lists based on the specified category(cat) and giving them a rank and title
+    input_list = sorted(input_list, key=lambda i: i[cat], reverse=True)
+    rk = 1
+    for item in input_list:
+        item['title'] = cat
+        item['rank'] = rk
+        rk += 1
+    return input_list
+
+
+def decide_top5_players(top5_list, cat_stat, category, first_name, last_name, player_data):
+    # Decides if a player's stat is better than the last guy's in the current top5_list
+    if len(top5_list) < 5:
+        top5_list.append({"season": player_data['season'], "first_name": first_name, "last_name": last_name, "pts": player_data['pts'],
+                          "reb": player_data['reb'], "ast": player_data['ast'], "stl": player_data['stl'], "blk": player_data['blk']})
+
+    else:
+        if cat_stat > top5_list[-1][category]:
+            top5_list.pop()
+            top5_list.append({"season": player_data['season'], "first_name": first_name, "last_name": last_name, "pts": player_data['pts'],
+                              "reb": player_data['reb'], "ast": player_data['ast'], "stl": player_data['stl'], "blk": player_data['blk']})
+
+    top5_list = top5_list_sort(category, top5_list, first_name, last_name)
+    return top5_list
+
+
+def parse_stat_by_category(category, cat_stat, first_name, last_name, player_data):
+    # use this function to see if a player's season average category is in the top5 of our current list
+    global top5_scorers, top5_rebounders, top5_assisters, top5_stealers, top5_blockers
```

Comment:
🚨 🚨 🚨 `global` is bad, evil, malicious, dangerous, maleficent, treacherous, and horrible 🚨 🚨 🚨 

Sometimes its hard but its really important that you create a structure that makes this unnecessary.